app/views/posts/show.html.erb

```diff
@@ -18,15 +18,20 @@
     <div class="post_txt">
         <p><%= @post.text %></p>
     </div>
+    <%= form_with model: @like, url: user_post_likes_path(@user, @post) do |f| %>
```

Comment:
- [ ] [REQUIRED] Hello @supreeti , you have done a great job so far, how ever, we seem to have an error while trying to create Likes. This error suggests that the link_to helper in your posts/show.html.erb view is trying to generate a URL for the create action of the LikesController without providing the required :user_id parameter.

Here's an example of how you might use link_to to create a like for a post, assuming you're on the post's show page:

``` ruby
<%= link_to "Like", user_post_likes_path(user_id: @post.author_id, post_id: @post.id), method: :post %>
```

In the example, I just stated above for you,` user_post_likes_path` generates the correct URL for the create action of the LikesController. user_id and post_id are passed as parameters to the route, ensuring that the required keys are provided.

Also, make sure you replace user_id: @post.author_id with the appropriate way of getting the user ID associated with the post. If `@post.author_id` is unavailable, replace it with the correct variable or method to obtain the user ID.

<img width="848" alt="Screenshot 2023-10-14 at 17 24 10" src="https://github.com/supreeti/BlogApp-ROR/assets/58692590/980bb310-570a-41ef-8295-a2e8bdcf6471">